giraffe/src/utils/fromFlux.ts

```diff
@@ -246,45 +246,68 @@ export const fastFromFlux = (fluxCSV: string): FromFluxResult => {
     let columnDefault: any = ''
 
     for (const [start, end] of chunks) {
-      const splittedChunk = fluxCSV.substring(start, end).split('\n')
-
-      const tableTexts = []
-      const annotationTexts = []
-
-      splittedChunk.forEach(line => {
-        if (line.startsWith('#')) {
-          annotationTexts.push(line)
+      annotationText = ''
+      tableText = ''
+
+      let index = start
+
+      while (index < end) {
+        const startsWithHash = fluxCSV[index] === '#'
+        const isPrevNewLine =
+          fluxCSV[index - 1] === '\n' || fluxCSV[index - 1] === undefined
+        // check to see if the one before that is whitespace or a new line
+        const isWhitespaceOrNewLine =
+          fluxCSV[index - 2] === undefined || fluxCSV[index - 2]?.trim() === ''
+        if (startsWithHash && isPrevNewLine && isWhitespaceOrNewLine) {
+          const nextIndex =
+            fluxCSV.substring(index, end).lastIndexOf('\n#') + index
+          const endOfAnnotation = fluxCSV.indexOf('\n', nextIndex + 1)
```

Comment:
this still doesn't consider whether we're in the middle of a string or if this is actually the end of the annotation text